docs/mail-merge/docs_mail_merge.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright ©2018-2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+docs-mail-merge.py (2.x or 3.x)
+
+Google Docs (REST) API mail-merge sample app
+"""
+from __future__ import print_function
+import time
+
+from googleapiclient import discovery
+from httplib2 import Http
+from oauth2client import file, client, tools
+
+# fill-in your template file ID
+DOCS_FILE_ID = 'YOUR_TMPL_DOC_FILE_ID'      # Docs template
+SHEETS_FILE_ID = 'YOUR_SHEET_DATA_FILE_ID'  # Sheets data source
+CREDS_FILE = 'credentials.json'
+SCOPES = (  # iterable or space-delimited string
+    'https://www.googleapis.com/auth/drive',
+    'https://www.googleapis.com/auth/documents',
+    'https://www.googleapis.com/auth/spreadsheets.readonly',
+)
+COLUMNS = ['to_name', 'to_title', 'to_company', 'to_address']
+OUTPUT = True # change to False to suppress output
+SOURCES = ('text', 'sheets')
+SOURCE = 'text' # or 'sheets' to change data source
+
+def get_http_client():
+    """Manage OAuth tokens in 'token.json' file given the downloaded
+       'credentials.json' file along with requested OAuth2 scopes.
+    """
+    store = file.Storage('token.json')
+    creds = store.get()
+    if not creds or creds.invalid:
+        flow = client.flow_from_clientsecrets(CREDS_FILE, SCOPES)
+        creds = tools.run_flow(flow, store)
+    return creds.authorize(Http())
+
+# create service endpoints to the 3 APIs
+HTTP = get_http_client()
+DRIVE = discovery.build('drive', 'v3', http=HTTP)
```

Comment:
I feel iffy about using all caps here since these aren't really constants:
https://www.python.org/dev/peps/pep-0008/#global-variable-names
Looks like it recommends lowercase, let's go with drive_service, etc.

Reply:
Technically you're correct because Python only supports system-level constants (`None`, `False`, `True`), however these values don't change value once assigned, so they're effectively constants where `ALLCAPS` is supported as per python.org/dev/peps/pep-0008/#id48 ... I'd like to reserve normal lowercase for variables that my code actually mutates. However, in the end I think I can be convinced otherwise, so I'll make the change if you insist. It's just easier-to-read when shorter & in CAPS IMO.